### DISCUSSION TITLE
Annotations?

### DISCUSSION THREAD

**Discussion by ashmitb95** [tradex-app/TradeX-chart#4] [General]
Is it possible to add annotations currently?

**Comment by tradex-app** [comment]
Currently no, but I'll add that to the list.

Technically, all of the methods already exist to build such a feature, they would needed to be coded together.

**Comment by tradex-app** [comment]
@ashmitb95 

I'm coming close to adding this feature.

I've found an example like this.
https://canvasjs.com/javascript-stockcharts/stockchart-annotation-indexlabel/

Do you have any other examples for me to consider?

**Reply by ashmitb95** [reply]
This is another you can look at:

https://www.infragistics.com/products/ignite-ui-react/react/components/charts/types/stock-chart

Check under annotations

**Reply by tradex-app** [reply]
Thank you Ashmit. I've taken a look. They don't look particularly clear and visible.
Here is what I am proposing for TradeX Chart.
Icons placed on the chart are clickable to reveal the annotation.
https://tradex-chart.guildmedia.net/reference/annotations/

Otherwise if you want text on the chart, then I would implement that as text drawing tool.

If you have any better examples or feedback, it is welcome.

**Reply by ashmitb95** [reply]
I recently found another library, but forgot to mention it. 

Have you checked this out? https://github.com/project-nv/night-vision

I found this one super interesting. It's an updated fork of the OG TradingVuejs

**Reply by tradex-app** [reply]
Yes, I'm in contact with the developer of Night Vision. It is a better and more modern version of his original  charting project, TradingVueJS. TradingVueJS is dependent upon Vue2 and since the developer abandoned the Vue3 version, this pushed me to start the TradeX Chart.

It would seem the developer has stopped working on Night Vision, according to his replies to questions on discord, as he expects the community to take over the development. So far that hasn't happened.

Night Vision does not have all of the features that TVJS has.

It took him two years to develop TradingVueJS. TradeX has been in development for a year and a half now, so perhaps it will take another six months to reach feature parity with TVJS or better.

If you are looking around, KlineChart is another one which is in active development. But there seems to be very little documentation for it, either in Chinese or English.

**Reply by tradex-app** [reply]
Here is the current development road map for TradeX Chart. It will give you some idea of where it is going.

https://tradex-chart.guildmedia.net/TradeX-chart-Development-Roadmap.pdf
